source/hud/blocklyButton.js

```diff
@@ -22,8 +22,16 @@ this.onClick = function() {
     var hud = this.parent;
     hud.lastBlocklyNodeID = hud.lastBlocklyNodeID || this.scene.player.rover.id;
     if ( hud.lastBlocklyNodeID !== undefined ) {
+        
+        var blocklyNode = blocklyNodes[ hud.lastBlocklyNodeID ];
+
+        if ( blocklyNode && Blockly.mainWorkspace ) {
+            Blockly.mainWorkspace.maxBlocks = blocklyNode.ramMax;
+        } else {
+            console.log("Either blocklyNode or Blockly.mainWorkspace are missing!");
+        }
```

Comment:
This block of code should be in `index.js` under the `satProperty` handler for `blockly_activeNodeID`.